src/main/java/com/group7/ecommerce/repository/ProductRepository.java

```diff
@@ -1,17 +1,46 @@
 package com.group7.ecommerce.repository;
 
+import com.group7.ecommerce.dto.request.ProductFilterDto;
+import com.group7.ecommerce.dto.response.ProductListItemResponse;
 import com.group7.ecommerce.dto.response.FeaturedProductView;
 import com.group7.ecommerce.entity.Product;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import com.group7.ecommerce.enums.OrderStatus;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
-public interface ProductRepository extends JpaRepository<Product, Long> {
+public interface ProductRepository extends JpaRepository<Product, Long>, ProductRepositoryCustom {
+
+	@Query(value = """
+		SELECT new com.group7.ecommerce.dto.response.ProductListItemResponse(
+			p.id,
+			p.name,
+			p.description,
+			p.sellingPrice,
+			pi.imageUrl,
+			CASE WHEN p.category IS NULL THEN NULL ELSE p.category.name END,
+			p.stockQuantity
+		)
+		FROM Product p
+		LEFT JOIN ProductImage pi ON pi.product = p AND pi.isPrimary = true
+		WHERE p.isDeleted = false
+		ORDER BY p.createdAt DESC
+		""",
+		countQuery = """
+		SELECT COUNT(p)
+		FROM Product p
+		WHERE p.isDeleted = false
```

Comment:
1. Cân nhắc dùng Spring Data JPA Specifications hoặc QueryDSL để viết điều kiện động, dễ bảo trì hơn


2. Sử dụng SELECT new ... để map sang DTO là đúng cách. Tuy nhiên, nếu DTO có nhiều trường, hoặc mapping phức tạp, có thể dùng projection interface (Spring 3.0+) để mapping trực tiếp sang interface, giúp code ngắn gọn hơn.

3. Sử dụng CASE WHEN p.category IS NULL THEN NULL ELSE p.category.name END ổn, nhưng nếu entity đã mapping quan hệ đúng, có thể dùng p.category?.name với projection interface